src/main/java/com/OrganizationController.java

```diff
@@ -1,54 +1,89 @@
-            IOException {
-        Organization organization = ServletUtil.expect( Organization.class, request);
-        ServletUtil.respond(organizationService.saveOrganization(organization), response);
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        try {
+            Organization organization = ServletUtil.expect(Organization.class, request);
+            ServletUtil.respond(organizationService.saveOrganization(organization), response);
+        } catch (BadRequestException | NotFoundException e) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            ErrorResource error = new ErrorResource("Bad Request", e.getMessage(), HttpServletResponse.SC_BAD_REQUEST);
+            ServletUtil.respond(error, response);
+        }
     }
 
-    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException,
-            IOException {
-        String id = request.getParameter("id");
-        if (id == null){
-            ServletUtil.respond(organizationService.getAll(), response);
-        }else{
-            Organization organization = organizationService.getOne(Integer.parseInt(request.getParameter("id")));
-
-            if(organization != null){
-                ServletUtil.respond(organizationService.getOne(Integer.parseInt(request.getParameter("id"))), response);
-            }else{
-                response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        try {
+            String id = request.getParameter("id");
+            if (id == null) {
+                ServletUtil.respond(organizationService.getAll(), response);
+            } else {
+                Organization organization = organizationService.getOne(Integer.parseInt(request.getParameter("id")));
+                ServletUtil.respond(organization, response);
             }
+        } catch (NotFoundException e) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            ErrorResource error = new ErrorResource("Not found", e.getMessage(), HttpServletResponse.SC_NOT_FOUND);
+            ServletUtil.respond(error, response);
         }
     }
 
     @Override
-    protected void doDelete(HttpServletRequest request, HttpServletResponse response) throws ServletException,
-            IOException {
-        int id = Integer.parseInt(request.getParameter("id"));
-        organizationService.deleteOrg(id);
+    protected void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        try {
+            String id = request.getParameter("id");
+            if (id == null) {
+                throw new BadRequestException("Organization doesn't exists for id:"+ id);
```

Comment:
This should throw a Bad request.
id == null means user haven't sent an ID. That means it is a bad request.